src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+
+    private int threshold;
+    private int size;
+    private Node<K,V>[] table;
+
+    private static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        size = getSize();
+        table = (Node<K,V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        Node<K,V> currentNode = table[index];
+        Node<K,V> nextNode = new Node<>(index, key, value, null);
+        if (currentNode == null) {
+            table[index] = nextNode;
+        } else {
+            while (true) {
+                if (key == currentNode.key || (key != null && key.equals(currentNode.key))) {
```

Comment:
let's move this checking to the while loop

Reply:
Decided to change the loop condition to check if the nextNode is not null, cause I think we need both 'if' checks in a loop